### PR TITLE
Add compatibility with namespaced PHPUnit

### PR DESCRIPTION
**Pull request by nicolas-grekas**
Replaces #106, fixes Bug-Debian: https://bugs.debian.org/882930

### CODE REVIEW

File: tests/Iconv/IconvTest.php [hunk 1]
```diff
@@ -106,11 +107,12 @@ public function testIconvMimeDecode()
 
     /**
      * @covers Symfony\Polyfill\Iconv\Iconv::iconv_mime_decode
+     * @expectedException \PHPUnit\Framework\Error\Notice
+     * @expectedExceptionMessage Detected an illegal character in input string
      */
     public function testIconvMimeDecodeIllegal()
     {
         iconv_mime_decode('Legal encoded-word: =?utf-8?Q?*?= .');
-        $this->setExpectedException('PHPUnit_Framework_Error_Notice', 'Detected an illegal character in input string');
```
Comment (stof) [inline tests/Iconv/IconvTest.php]: this test should be split in 2, so that the exception is allowed only in the second call, not the first one.

Comment (stof) [reply tests/Iconv/IconvTest.php]: or it should use a method_exists check to keep configuring it inline

File: tests/Iconv/IconvTest.php [hunk 2]
```diff
@@ -106,11 +107,12 @@ public function testIconvMimeDecode()
 
     /**
      * @covers Symfony\Polyfill\Iconv\Iconv::iconv_mime_decode
+     * @expectedException \PHPUnit\Framework\Error\Notice
```
Comment (stof) [inline tests/Iconv/IconvTest.php]: does this work even on older versions where the class name had underscores ?

File: tests/Intl/Grapheme/GraphemeTest.php [hunk 3]
```diff
@@ -11,22 +11,24 @@
...
+class GraphemeTest extends TestCase
 {
     /**
      * @covers Symfony\Polyfill\Intl\Grapheme\Grapheme::grapheme_extract
+     * @expectedException \PHPUnit\Framework\Error\Warning
+     * @expectedExceptionMessage expects parameter 1 to be string, array given
      */
     public function testGraphemeExtractArrayError()
     {
         grapheme_extract('', 0);
```
Comment (stof) [inline tests/Intl/Grapheme/GraphemeTest.php]: why do we have this call here ?

**Comment by nicolas-grekas** [comment]
Now green

**Pull request merged**